models/chainlink/arbitrum/chainlink_arbitrum_ccip_reverted_transactions.sql

```diff
@@ -0,0 +1,58 @@
+{{
+  config(
+    
+    alias='ccip_reverted_transactions',
+    partition_by=['date_month'],
+    materialized='incremental',
+    file_format='delta',
+    incremental_strategy='merge',
+    unique_key=['tx_hash', 'node_address']
+  )
+}}
+
+{% set incremental_interval = '7' %}
+
+WITH
+  arbitrum_usd AS (
+    SELECT
+      minute as block_time,
+      price as usd_amount
+    FROM
+      {{ source('prices', 'usd') }} price
+    WHERE
+      symbol = 'ETH'
+      {% if is_incremental() %}
+        AND minute >= date_trunc('day', now() - interval '{{incremental_interval}}' day)
+      {% endif %}      
+  ),
+  ccip_reverted_transactions AS (
+    SELECT
+      tx.tx_hash as tx_hash,
+      MAX(tx.block_time) as block_time,
+      cast(date_trunc('month', MAX(tx.block_time)) as date) as date_month,
+      tx."from" as "node_address",
+      MAX((cast((gas_used) as double) / 1e18) * effective_gas_price) as token_amount,
+      MAX(arbitrum_usd.usd_amount) as usd_amount
+    FROM
+      {{ ref('chainlink_arbitrum_ccip_send_traces') }} tx
+      LEFT JOIN {{ source('arbitrum', 'transactions') }} tx2 ON tx2.hash = tx.tx_hash
```

Comment:
can you add incremental filter on this base table join condition?

also, if possible, an inner join would help performance